sst.config.ts

```diff
@@ -31,6 +31,14 @@ function createLambdaFunction() {
     NODE_ENV: process.env.NODE_ENV ?? 'staging',
     ANTEATER_API_KEY: process.env.ANTEATER_API_KEY!,
     EXTERNAL_USER_READ_SECRET: process.env.EXTERNAL_USER_READ_SECRET!,
+    OTEL_EXPORTER_OTLP_HEADERS: process.env.OTEL_EXPORTER_OTLP_HEADERS!,
+    // hardcoded OTEL options
+    ...($app.stage === 'prod' && {
+      AWS_LAMBDA_EXEC_WRAPPER: '/opt/otel-handler',
+      NODE_OPTIONS: '--require @opentelemetry/auto-instrumentations-node/register',
+      OTEL_SERVICE_NAME: 'peterportal-backend',
+      OTEL_EXPORTER_OTLP_ENDPOINT: 'https://ingress.us-west-2.aws.dash0.com',
+    }),
```

Comment:
Not quite sure if I like this as a spread operator as opposed to a separate if statement and `Object.assign`, but otherwise this looks fine.